torchtext/experimental/vocab.py

```diff
@@ -1,12 +1,57 @@
+from collections import OrderedDict
 import logging
 from typing import Dict, List
 
 import torch
 import torch.nn as nn
+from tqdm import tqdm
 
 logger = logging.getLogger(__name__)
 
 
+def _infer_shape(f):
+    num_lines = 0
+    for line in f:
+        num_lines += 1
+    f.seek(0)
+    return num_lines
+
+
+def vocab_from_file_object(file_like_object, **kwargs):
+    r"""Create a `Vocab` object from a csv file like object.
```

Comment:
I'd separate the "csv" and "file like object" part. I'd say this accepts a file like object and then describe the data protocol which is separate lines in csv format.

Reply:
Done!